### PR TITLE
Tensor Product Operator

### PR DESCRIPTION
**Pull request by vpuri3**
close https://github.com/SciML/SciMLOperators.jl/issues/23

### CODE REVIEW

**Comment by vpuri3** [comment]
done with functionality. just needs tests now

**Comment by vpuri3** [comment]
done

**Comment by vpuri3** [comment]
i dont think there are any integration tests. @ChrisRackauckas could you merge and version bump? id like to start using these operators here: https://github.com/vpuri3/PDEInterfaces.jl

File: src/SciMLOperators.jl [hunk 1]
```diff
@@ -42,7 +45,9 @@ export ScalarOperator,
        MatrixOperator,
        DiagonalOperator,
        AffineOperator,
-       FunctionOperator
+       FunctionOperator,
+       TensorProductOperator,
+       ⊗
```
Comment (ChrisRackauckas) [inline src/SciMLOperators.jl:50]: I'm not sure about exporting this operator. Other than that, I think this is good to go.

Comment (vpuri3) [reply src/SciMLOperators.jl:50]: do you mean just the `\otimes` overload or TensorProductOperator?

Comment (vpuri3) [reply src/SciMLOperators.jl:50]: or both?

Comment (vpuri3) [reply src/SciMLOperators.jl]: either way i took out the export

**Comment by vpuri3** [comment]
good to go

File: src/SciMLOperators.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/SciMLOperators.jl]: Just the `⊗`

Comment (vpuri3) [reply src/SciMLOperators.jl]: ok lemme add the TensorProductOperator export back

Comment (vpuri3) [reply src/SciMLOperators.jl]: done

**Comment by vpuri3** [comment]
@ChrisRackauckas could you version bump?

**Pull request merged**